### ISSUE TITLE
Toggle quest filters

### ISSUE THREAD

**Issue by waldyrious** [westnordost/StreetComplete#1301]
**Use case**
Sometimes I'm passing through an area where I've filled all of the quests I chose to make visible, and I'd like the app to show quests I've filtered out, but don't want to make that change permanent.

**Proposed Solution**
A button to toggle quest filtering on/off would be nice to do this without completely resetting the quest selection and ordering configured in the settings.

**Comment by matkoniecz** [comment]
On one hand it is something that I like and one of reasons for maintaining a personal version of this program for my own use.

On the other hand it is impossible to implement all functionality that can be useful for somebody and keep app simple (AKA feature creep).

**Comment by westnordost** [comment]
For what reason do you switch on and off the quests but do not want that change to be permanent?

**Comment by waldyrious** [comment]
@westnordost I am not sure I understand your question. I didn't mean to toggle quests off, but to toggle quest selection/filtering off (i.e. temporarily show all available quests, without losing the customization I had made in the settings).

**Comment by westnordost** [comment]
But why do you do that?

**Comment by waldyrious** [comment]
> But why do you do that?

I mentioned that in the opening comment:

> **Use case**
> Sometimes I'm passing through an area where I've filled all of the quests I chose to make visible, and I'd like the app to show quests I've filtered out, but don't want to make that change permanent.

Perhaps I worded that confusingly? I can try to rephrase or expand if that would help.

**Comment by matkoniecz** [comment]
It is a bit unclear because solved quests disappear.

What your exactly mean by 

>  I've filled all of the quests I chose to make visible

? Both solving and entering quest and deciding to hide them makes them hidden.

**Comment by westnordost** [comment]
@waldyrious Quests are generally displayed in the priority order (you) specified in the settings. For example, the "do these traffic lights have a button for pedestrians to request green light?" has a very low priority, "what is the name of this road?" has the highest priority. That means that even if the traffic lights button quest is enabled, it will only ever be downloaded and displayed if there are no more important quests around.
So, if you find certain quests very unimportant and you do not usually want to solve them if any you are interested in are around, which seems to be your use case here, then you can drag them further down the list and you are fine. No need to toggle enabling/disabling them.

**Comment by waldyrious** [comment]
Thanks for that perspective, @westnordost, indeed I hadn't thought about it that way. After some consideration, I realize now that the need I felt for toggling filters came from two unrelated issues:

The first one is that the way the app selects which quests to show, depending on the zoom level and the sorting/filtering settings, sometimes seems to behave in unexpected ways. If I recall correctly (though I haven't tested that well enough to make definitive conclusions), in my experience some quests appear ahead of others even if I set the latter with a higher priority, so I had to actually turn some quests off in order to make the other ones visible. (Perhaps due to zooming levels or size of the features? I'd need to test more exhaustively to be sure.)

The second issue is that I use StreetComplete in different contexts, including ones where some quests are almost impossible to complete. For example, I often travel as a passenger in a car or bus, where I can't really read opening hours of businesses, and even door numbers are very hard because the GPS position updates in jumps and I can't be sure what building in the map corresponds to the one I had read the door number from as the vehicle zoomed past it. On other cases, say I'm walking home from work, and then I can complete some more quests, but not those that would require me to delay my journey. And then there's times where I actually go out to map, and can afford to go back and forth, stop to read opening hours, signs, and so on (but this much less common).

So ideally there would be multiple presets one could configure for such contexts, with different quest filtering/ordering configurations. When walking I'd want higher priority to the quests I can't complete while on a bus, and when I'm on a bus I want the opposite (and so on for other contexts).

Now, if I were able to simply toggle between casual mapping mode and dedicated surveying mode (where all quests would be visible), I believe that would alleviate these issues. However, I am aware that that wouldn't actually be solving the core of those problems, so I'd understand if you'd prefer to tackle the underlying issues rather than implementing this toggle. I just think it could be a good trade-off especially for the "two mapping modes" problem, since I assume it's much easier to implement than multiple quest filtering presets.

**Comment by matkoniecz** [comment]
> in my experience some quests appear ahead of others even if I set the latter with a higher priority

What is usually happening is that on say zooming out high priority quest marker blocks middle priority closer quest labels - what may unblock some low priority quest. It may give an illusion of low priority quests blocking middle priority quests.

If higher priority quests are actually blocked by one listed later- than it is probably unintended and bug.

**Comment by waldyrious** [comment]
Yeah, I thought it could be something like that. I'll make sure to take note of the details next time I come across that.

**Comment by westnordost** [comment]
> The second issue is that I use StreetComplete in different contexts, including ones where some quests are almost impossible to complete. For example, I often travel as a passenger in a car or bus, where I can't really read opening hours of businesses, and even door numbers are very hard because the GPS position updates in jumps and I can't be sure what building in the map corresponds to the one I had read the door number from as the vehicle zoomed past it. On other cases, say I'm walking home from work, and then I can complete some more quests, but not those that would require me to delay my journey. And then there's times where I actually go out to map, and can afford to go back and forth, stop to read opening hours, signs, and so on (but this much less common).

I have to confess that I was suspecting that this was the "hidden" use case/feature request from the start ;-).

This is a valid feature request, though it should be put somewhere where it does not clutter up the UI but still be more visible than the quest selection screen (or part of it??), to not conflict in a way with that screen and be clear to the user how the two interact.
If you have some input or thoughts on how these profiles could be designed to be all that, it is very welcome.

**In any case, could you post this as a new feature request and close this ticket? This way it is clearer what to do.**

**Comment by waldyrious** [comment]
Ahaha, you figured me out then, even before I did :)

Yeah, I will submit a new issue then, later today (hopefully with some ideas for the UI), and I'll close this one now.

**Comment by rugk** [comment]
Regarding the end use-case: This was partially discussed in https://github.com/westnordost/StreetComplete/issues/1203, too, altghough (initially, at least) a different solution has been proposed there.... 😄 

And it has finally been implemented and the "deselect all" button should have solved this use-case.

**Comment by waldyrious** [comment]
Apologies for never following up with this. Looks like @Helium314 took the lead and opened #1654 which is exactly what I needed.
